cpp/src/KIM_CollectionsImplementation.cpp

```diff
@@ -76,148 +80,38 @@
 
 namespace
 {
-void EraseTrailingNulls(std::string & str)
-{
-  size_t first = str.find_first_of('\0');
-  if (first != std::string::npos) str.erase(first, std::string::npos);
-}
-
 typedef std::map<KIM::CollectionItemType,
-                 std::string,
+                 KIM::FILESYSTEM::PathList,
                  KIM::COLLECTION_ITEM_TYPE::Comparator>
     ItemTypeToStringMap;
 
-std::string LibraryName(KIM::CollectionItemType const itemType,
-                        std::string const & path,
-                        std::string const & name)
+KIM::FILESYSTEM::Path LibraryName(KIM::CollectionItemType const itemType,
+                                  KIM::FILESYSTEM::Path const & path)
 {
-  std::string libName(path);
-
-  libName += "/" + name + "/" KIM_SHARED_MODULE_PREFIX + KIM_PROJECT_NAME + "-";
+  KIM::FILESYSTEM::Path libName = path / KIM_SHARED_MODULE_PREFIX;
+  libName += KIM_PROJECT_NAME;
+  libName += "-";
 
-  {
-    using namespace KIM::COLLECTION_ITEM_TYPE;
-    if (itemType == modelDriver) { libName += KIM_MODEL_DRIVER_IDENTIFIER; }
-    else if (itemType == portableModel)
-    {
-      libName += KIM_PORTABLE_MODEL_IDENTIFIER;
-    }
-    else if (itemType == simulatorModel)
-    {
-      libName += KIM_SIMULATOR_MODEL_IDENTIFIER;
-    }
-    else
-    {
-      libName += "UNKNOWN-COLLECTION-ITEM-TYPE";
-    }
-  }
+  using namespace KIM::COLLECTION_ITEM_TYPE;
```

Comment:
I'm not sure what you mean. A `using namespace` directive inside a function scope will not pollute the surrounding namespace, because it is limited to the function scope. Restricting the scope even further seems extreme, especially if the function is short like this one. Or is that required by your style?

Reply:
Sorry; this is just me being overly pedantic (Indeed I was thinking about the namespace within the function, but as you say it is too short to really worry about).  It is fine.  Let's leave it as is.